### DISCUSSION TITLE
(Long Term) I read some very interesting things on your medium blog, Congratulations on Lokinet, and a few concerns.

### DISCUSSION THREAD

**Discussion by Destroyinator69420** [oxen-io/lokinet#1836] [General]
I would like to start by thanking you for developing Lokinet, it has become a valuable tool in taking back my privacy. I consider Lokinet to be best overall in the desktop. I would like to offer the team a few ideas as a show of thanks.

I read on the oxen medium blog that all service nodes will eventually become exit nodes, I worry that this may discourage future service node operators and investors when it is implemented, but I think that exit node operators are the bravest of all mixnet relay operators, since they risk legal trouble with copyright trolls and police so that they can help us all access the internet as it was originally intended. I think that for their bravery and sacrifice for a freer the internet, the oxen team might want to think of a separate Lokinet block reward for exit nodes, the trade off they pay off the return on investment and reach profitability faster, and in exchange they help Lokinet grow its userbase, my first introduction with the oxen ecosystem was with Lokinet back when oxen was called Loki. I also worry about the prospect of an exit node marketplace, since unequal access to a free internet is a problem which needs to be solved, and paid exit nodes segregate exit nodes, if implemented, I worry it could put strain on the free exit nodes by separating customers who are willing to pay for less exit congestion from the non paying users who have to deal with it. If all exit nodes were free, but were paid a second block reward, the network congestion could be spread out over many more exit nodes competing for the exit node block reward, especially if their was a randomized option for setting exit nodes. I think this could be done by having exit nodes list themselves as exits on the oxen blockchain, for the purpose of burning oxen, it would be prudent to have service nodes identify themselves with a oxen name system name following the same format the public exit nodes are already using. (exit_name-exit.loki)

I also worry that since smartphones are becoming ever more prevalent, that Lokinet for mobile is more important than ever, since Lokinet is one of the few apps with VPN functionality that is fast, free, and anonymous, which give it the perfect properties for a mixnet/VPN app, which android is sadly lacking and if implemented would grow the lokinet community by bringing in new users, I was reading a while ago a large minority of smartphone users don't own a laptop or desktop PC, which is frightening to say the least.

The next concern I have for the growth of Lokinet over the time is file sharing, since the bittorrent protocol is only supported over exits, the ability to seed hidden files to hidden feeders would make Lokinet much more worthwhile than I2P for torrenting, but Libtorrent would have to be changed so that there is a configuration that would change IP addresses to the temporary addresses given to users for hidden seeding only, and support would have to be maintained for seeding through exits that allow it. The real deal though is whether these changes make it back to the official Libtorrent, although I think that torrenting is less of an issue to be solved right now than increasing the diversity of exits, since having only four public exits at the time of writing this is probably makes Lokinet users much easier to track.

Thank you and happy new year,
destroyer

**Comment by jagerman** [comment]
Hey Destroyer, 
Thanks for reaching out. Sorry for the delay in reply; I drafted most of this several days ago but other things came up and I forgot about it for a few days.  My apologies.

Firstly, I appreciate how much thought you’ve put into Lokinet and its design — community feedback is always welcome, especially when it’s so thorough. 

> I read on the oxen medium blog that all service nodes will eventually become exit nodes

In Lokinet’s early life, we did propose that all the nodes in the network could be used exits. However, we moved away from this idea some time ago — mostly due to issues which you’ve already raised. 

> I think that for their bravery and sacrifice for a freer the internet, the oxen team might want to think of a separate Lokinet block reward for exit nodes

Exit operators probably _do_ need additional incentives to outweigh the risks and demands that come with running an exit node—although altruism can also be a factor, as it is in networks like Tor’s—but we don’t think changing the block reward is the best way to achieve this. 

If all exit nodes are receiving the same reward for their service, the underlying financial incentive will be to spend as few resources as possible to meet the minimum requirements for an exit — resulting in the network having lower quality exits on-average. An exit marketplace ties the incentive directly to the quality of the service the exit can provide — whether that be bandwidth, lower latency, uptime, or something else which improves the user’s experience. 

> pay off the return on investment and reach profitability faster

If we do give exit operators a bigger reward _without_ increasing the staking requirements, this will incentivise attempts to cheat the system. 

> … since unequal access to a free internet is a problem which needs to be solved

It certainly is — but whether that’s an issue that can be wholly solved via free exit nodes is another matter. The problem with providing free exit node access is that they attract a lot of ‘bad’ traffic—from things like torrenting, malware, and port scanning—which makes them unusable for regular users. 

Despite relative obscurity, we’ve already seen this issue with the exit node run by our Foundation (exit.loki). As the network becomes more well-known and exit nodes become more numerous, this issue would no doubt be greatly exacerbated. 

> If all exit nodes were free, but were paid a second block reward, the network congestion could be spread out over many more exit nodes competing for the exit node block reward

As much as the network has grown (more than 1700 service nodes now for several months), even with a high proportion of the network operating as free exit nodes — it would still likely become congested by the traffic types I mentioned above. 

Without creating a strong disincentive against spamming the network with those kinds of traffic, it makes the exit capacity vulnerable to abuse or attack. The result: regular users have their service significantly degraded.

On the whole, having a system which allows individual nodes to distinguish themselves based on quality of service gives both operators and users more agency and control. 

> I also worry that since smartphones are becoming ever more prevalent, that Lokinet for mobile is more important than ever

Agreed, and mobile versions of Lokinet have been on the agenda for a long time (we have a working Android version in testing now for a while; iOS is likely to be a bigger obstacle, unfortunately). We do want to deliver Lokinet to mobile devices, though, and the increasing use and reliance on mobile devices is very concerning when juxtaposed with their relative lack of security and privacy. 

Generally speaking, mobile operating systems (*especially* Apple's) take a ‘walled garden’ approach to their design, making it harder for tools like Lokinet to work properly. 

Having said that, it _is_ currently possible to use Lokinet on Android, albeit it’s not neatly packaged in the Play Store. That’s something we hope to do in the future, though. 

Unfortunately, iOS is a much more difficult case to solve — and that one will be an ongoing challenge for us to develop. 

> The next concern I have for the growth of Lokinet over the time is file sharing

Torrenting through Lokinet, while possible (as most things are through Lokinet), isn’t really part of Lokinet’s intended use case. We want people to be able to freely and privately access information on the internet, but peer to peer file-sharing isn’t necessarily required to achieve that and would put a great strain on the network.  We aren't doing anything actively against it, but it's not our top priority because of the tendency it has to suck up considerable amounts of traffic to the exclusion of other uses.
